Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -234,7 +234,7 @@ namespace AZ
         {
             ShaderReloadDebugTracker::ScopedSection reloadSection("{%p}->Material::OnAssetReloaded %s", this, asset.GetHint().c_str());
 
-            Data::Asset<MaterialAsset> newMaterialAsset = { asset.GetAs<MaterialAsset>(), AZ::Data::AssetLoadBehavior::PreLoad };
+            Data::Asset<MaterialAsset> newMaterialAsset = Data::static_pointer_cast<MaterialAsset>(asset);
```

Comment:
I think the casting happens under the hood when assigning from AZ::Data::Asset<T1> to AZ::Data::Asset<T2>. I have not seen Data::static_pointer_cast used before in these cases. Usually just direct assignment.

```
Data::Asset<MaterialAsset> newMaterialAsset = asset;
Data::Asset<StreamingImageAsset> streamingImageAsset = imageAsset;
Data::Asset<PassAsset> passAsset = asset;

```
and so on.

Reply:
Assuming Data::Asset<AssetData> asset;
In VC++ it's ok to do:
Data::Asset<MaterialAsset> newMaterialAsset = asset;
But CLang doesn't like that, and you would need to do
Data::Asset<MaterialAsset> newMaterialAsset ={ asset.GetAs<MaterialAsset>(), AZ::Data::AssetLoadBehavior::PreLoad };
Or even better (because you don't have to come up with a new LoadBehavior):
 Data::Asset<MaterialAsset> newMaterialAsset = Data::static_pointer_cast<MaterialAsset>(asset);

Reply:
Interesting. We do this all over the place in code that has to compile on clang. But it's just a recommendation for consistency. 
```
        void MaterialComponentController::InitializeMaterialInstance(const Data::Asset<Data::AssetData>& asset)
        {
            bool allReady = true;
            auto updateAsset = [&](AZ::Data::Asset<AZ::RPI::MaterialAsset>& materialAsset)
            {
                if (materialAsset.GetId() == asset.GetId())
                {
                    materialAsset = asset;
                }
```

```
        void ReflectionProbeComponentController::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
        {
            if (m_configuration.m_useBakedCubemap)
            {
                m_configuration.m_bakedCubeMapAsset = asset;
            }
            else
            {
                m_configuration.m_authoredCubeMapAsset = asset;
            }
        }
```